src/test/java/com/youengineering/openapi/GenerateTest.java

```diff
@@ -69,16 +70,52 @@ private GenerateMojo createGenerateMojo(Framework framework) throws Exception {
     private void assetGeneratedFile_Json(Framework framework) throws Exception {
         JsonNode actualNode = Json.mapper().readTree(getOpenAPIFile(framework, "generated", OutputFormat.json));
         JsonNode expectedNode = Json.mapper().readTree(getOpenAPIFile(framework, "expected", OutputFormat.json));
-        assertEquals(expectedNode, actualNode);
+
+        // Removing all null attributes as openApi deserializer user node.get("extensions") to get value.
+        // And if any of the vales are null, it is getting treated as "null" (String type) and failing with class cast exception
+        removeNullsFromJsonNode(actualNode);
+        removeNullsFromJsonNode(expectedNode);
+        OpenAPI actualOpenAPI = openAPIV3Parser.parseJsonNode("", actualNode).getOpenAPI();
+        OpenAPI expectedOpenAPI = openAPIV3Parser.parseJsonNode("", expectedNode).getOpenAPI();
+        assertOpenApis(actualOpenAPI, expectedOpenAPI);
     }
 
     private void assetGeneratedFile_Yaml(Framework framework) {
-        OpenAPI actualOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "generated", OutputFormat.yaml).getPath());
-        OpenAPI expectedOpenAPI = openAPIV3Parser.read(getOpenAPIFile(framework, "expected", OutputFormat.yaml).getPath());
+        OutputFormat yaml = OutputFormat.yaml;
```

Comment:
Why move the format to a constant here, but not for the JSON part? Personally I would keep it inline, as there is no real gain from moving an enum to a constant with the same name, but the most important part is to have it consistent. But it's a very minor thing, just wanted to mention it.

Reply:
To be honest I do know. I was playing a bit around here, Not sure what was in my mind.
I reverted. Good catch.

Reply:
> I will leave the final approval decision of what will be merged to @dominik-erni - When the approval of him exists, I can merge this PR and release the new version to Maven central, because if I remember correctly, currently only my ossrh user (carlkuesters) has the permission to do so. I will check the process of hwo to add Dominik and/or Marc from youEngineering to it as well. Would this be ok?

Yes, that is perfectly fine. Thanks a lot for the support.